app/src/main/java/com/everest/mydetails/HomeActivity.kt

```diff
@@ -0,0 +1,73 @@
+package com.everest.mydetails
+
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.os.Bundle
+import android.text.TextUtils
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+
+
+class HomeActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_first)
+
+        val colorDrawable = ColorDrawable(Color.parseColor("#0F9D58"))
+        supportActionBar?.setBackgroundDrawable(colorDrawable)
+
+        val submitButton = findViewById<Button>(R.id.submitButton)
+        val firstName = findViewById<EditText>(R.id.firstNameET)
+        val age = findViewById<EditText>(R.id.ageET)
+        val bio = findViewById<EditText>(R.id.bioET)
+        val phoneNumber = findViewById<EditText>(R.id.phoneNumberET)
+
+        submitButton.setOnClickListener() {
+            val intent = Intent(this, MainActivity::class.java)
+            if (isInputValid(firstName, age, bio, phoneNumber)) {
+                intent.putExtra(FIRST_NAME, firstName.text.toString())
+                intent.putExtra(AGE, age.text.toString())
+                intent.putExtra(BIO, bio.text.toString())
+                intent.putExtra(PHONENUMBER, phoneNumber.text.toString())
+                startActivity(intent);
+            }
+        }
+    }
+
+    private fun isInputValid(
+        firstName: EditText,
+        age: EditText,
+        bio: EditText,
+        phoneNumber: EditText
+    ): Boolean {
+        when {
+            TextUtils.isEmpty(firstName.text) or TextUtils.isEmpty(age.text) or TextUtils.isEmpty(
+                bio.text
+            ) or TextUtils.isEmpty(phoneNumber.text) -> {
```

Comment:
TextUtils.isEmpty(firstName.text) or TextUtils.isEmpty(age.text) or TextUtils.isEmpty(
                bio.text
            ) or TextUtils.isEmpty(phoneNumber.text)
All these implies something. So we can extract it into a method which returns boolean. for ex: `isDataFilled()` method